contracts/rust/deployer/src/builder.rs

```diff
@@ -235,19 +238,40 @@ impl<P: Provider + WalletProvider> DeployerArgs<P> {
                 // NOTE: we don't transfer ownership to multisig, we only do so after V2 upgrade
             },
             Contract::StakeTableV2 => {
-                crate::upgrade_stake_table_v2(provider, contracts).await?;
-
-                if let Some(multisig) = self.multisig {
-                    let stake_table_proxy = contracts
-                        .address(Contract::StakeTableProxy)
-                        .expect("fail to get StakeTableProxy address");
-                    crate::transfer_ownership(
+                let use_multisig = self.use_multisig;
+                let dry_run = self.dry_run;
+                tracing::info!(?dry_run, ?use_multisig, "Upgrading to StakeTableV2 with ");
+                if use_multisig {
+                    crate::upgrade_stake_table_v2_multisig_owner(
+                        provider,
+                        contracts,
+                        self.rpc_url.clone(),
+                        self.multisig.unwrap(),
```

Comment:
Are we checking anywhere that these unwraps won't fail? If not, it would be nice to have a descriptive error message instead of just a panic.

Reply:
good point, updated https://github.com/EspressoSystems/espresso-network/commit/7fc3e3dac250a3e854a4d1796e01a280b5cdc736